indigo_app/static/javascript/indigo/views/annotations.js

```diff
@@ -474,16 +476,17 @@
     },
 
     layout: function() {
-      this.gutter.runLayout();
+      this.gutter.layoutItems();
     },
 
     threadDeleted: function(view) {
       this.threadViews = _.without(this.threadViews, view);
       this.visibleThreads = _.without(this.visibleThreads, view);
-      var ix = this.gutter.items.indexOf(view);
-      if (ix > -1) {
-        this.gutter.items.splice(ix, 1);
-      }
+      // var ix = this.gutter.items.indexOf(view);
+      // if (ix > -1) {
+      //   this.gutter.items.splice(ix, 1);
+      // }
+      view.el.remove();
       this.layout();
```

Comment:
I suspect we can remove `this.layout()` both here and everywhere. The gutter should know when to layout.